File: 프로그래머스/2/42747. H－Index/H－Index.swift
import Foundation

func solution(_ citations:[Int]) -> Int {
    var citation = citations.sorted(by: >)
    
    var h = 0
    for i in 0..<citation.count {
        if citation[i] >= i + 1 { h = i + 1}
        else {break}
    }
    return h
}

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    int answer = citations.size();
    sort(citations.begin(), citations.end(), greater<>());
    for(int i=0;i<citations.size();i++) {
        if (citations[i] < i+1) {
            return i;
        }
    }
    return answer;
}